Testing : Improving Section & Complete documentation (#1507)
PrestaShop/docs: testing/integration-tests/_index.md

---
title: Integration tests
chapter: true
---

# Introduction

> Unit tests can validate the behavior of a php class when it can be isolated.

Reviewer: ```suggestion
Unit tests can validate the behavior of a PHP class when it can be isolated.
```

> However, some classes cannot be validated this way. Moreover, a lot of logic from PrestaShop is written into complex SQL queries that cannot be validated by those kind of tests. This is why we also need integration tests.

Reviewer: ```suggestion
However, some classes cannot be validated this way. Moreover, a lot of logic from PrestaShop is written into complex SQL queries that this kind of test cannot validate. This is why we also need integration tests.
```

We use 2 technologies for the integration tests in the `Integration` folder:

- [Behat](https://behat.org) for tests that are meaningful scenarios from a user point of view

> - [PHPUnit](https://phpunit.de) for tests which rather answer the need to test the technical behavior of a class or a component

Reviewer: ```suggestion
- [PHPUnit](https://phpunit.de) for tests that instead answer the need to test the technical behavior of a class or a component
```

## Stack

> We use the following stack:

Reviewer: This is a duplicate of the previous section

* [Behat](https://behat.org)
* [PHPUnit](https://phpunit.de) as test framework

> # Execute & Create tests

Reviewer 2: Hi, should be an H2 as there is already an H1

```suggestion
## Execute & Create tests
```
